src/hooks/useItems.ts

```diff
@@ -0,0 +1,74 @@
+import { useState } from 'react';
+import { client } from '../utils/fetchClient';
+
+export const useItems = <T extends { id : number }>() => {
+  const [items, setItems] = useState<T[] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const handleItemsFetch = async (url: string, id: number, msg: string) => {
+    setItems(null);
+    setErrorMsg('');
+    setLoading(true);
+
+    try {
+      setItems(await client.get(`${url}${id}`));
+
+      setLoading(false);
```

Comment:
Consider to move `setLoading(false)` to `finally` block to avoid code duplication. It will executed regardless of result